### PR TITLE
Enable nullable checks for metadata

### PR DESCRIPTION
**Pull request by yufeih**
#5515 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5573)

### CODE REVIEW

**Comment by mingwli** [comment]
Cache is being weird :P

**Comment by yufeih** [comment]
> Cache is being weird :P

This is probably due to my change

File: docs/specs/git.yml [hunk 1]
```diff
@@ -552,6 +552,30 @@ outputs:
       "updated_at": "2018-10-30 12:00 AM"
     }
 ---
+# Do not resolve contributors if repo and editRepositoryUrl are not GitHub repository
```
Comment (live1206) [inline docs/specs/git.yml]: could this be a separated PR?

Comment (yufeih) [reply docs/specs/git.yml]: This bug is actually detected by nullability check, see https://github.com/dotnet/docfx/pull/5573/files#diff-63c67683598f3abbc11bdf942d99d29eL82, the old implementation assumes `editRepositoryUrl` is correct and can potentially leak `null` to `GitHubAccessor`, after enable nullable check, the compiler help us detect this, so I'd like to include it in this PR to illustrate how nullable check helps write robust code.

**Pull request merged**